### ISSUE TITLE
Sqlite fails

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-backend#486]
There's a problem in the develop branch that causes Travis to fail in all PRs to the Backend develop branch. 
It was first spotted in #482 but that PR was merged anyway since it was more important to fix the memory leak.
It only occurs when running tests against SQLite. Both MySQL and PostgreSQL run fine.
It seems to be independent of Perl versions.

Here's what it looks like:
```
t/test_validate_syntax.t .. ok
t/test01.t ................ 1/? Use of uninitialized value in -e at /home/travis/build/zonemaster/zonemaster-backend/build_dir/blib/lib/Zonemaster/Backend/TestAgent.pm line 41.
t/test01.t ................ 11/? Use of uninitialized value in concatenation (.) or string at /home/travis/build/zonemaster/zonemaster-backend/build_dir/blib/lib/Zonemaster/Backend/RPCAPI.pm line 387.
Use of uninitialized value in -e at /home/travis/build/zonemaster/zonemaster-backend/build_dir/blib/lib/Zonemaster/Backend/TestAgent.pm line 41.
External query for afnic.fr, SOA attempted to m.root-servers.net/202.12.27.33 while running with no_network at /home/travis/build/zonemaster/zonemaster-backend/build_dir/blib/lib/Zonemaster/Engine/Recursor.pm line 182.
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 9 just after 22.
t/test01.t ................ Dubious, test returned 9 (wstat 2304, 0x900)
All 22 subtests passed 
Test Summary Report
-------------------
t/test01.t              (Wstat: 2304 Tests: 22 Failed: 0)
  Non-zero exit status: 9
  Parse errors: No plan found in TAP output
Files=2, Tests=49,  5 wallclock secs ( 0.02 usr  0.01 sys +  4.62 cusr  0.13 csys =  4.78 CPU)
Result: FAIL
```

To reproduce in Travis, just make a dummy PR against the develop branch.

**Comment by mattias-p** [comment]
The problem is not limited to those PRs. The problem is in the develop branch so it affects all PRs against the develop branch. It was introduced in #482.

**Comment by matsduf** [comment]
Please create a better issue. I just wanted to create something so that we do not forget about the problem.

**Comment by mattias-p** [comment]
I'll update the description of this one.

**Comment by matsduf** [comment]
@mattias-p, please go ahead!

**Comment by mattias-p** [comment]
@matsduf Done.

**Comment by mattias-p** [comment]
This affects installation using `cpanm` on CentOS. (haven't tested any others)
It's possible to work around the problem using `cpanm --notest`.

**Comment by mtoma** [comment]
This seems quite clear to me:
External query for afnic.fr, SOA attempted to m.root-servers.net/2001:dc3::35 while running with no_network at /home/travis/build/zonemaster/zonemaster-backend/build_dir/blib/lib/Zonemaster/Engine/Recursor.pm

This is a croak in the engine.

I can't reproduce this issue in the test VM neither with export ZONEMASTER_RECORD=0 nor without.
I see the test01.t file was modified on the VM but I don't know what was modified.

@vlevigneron any idea why the engine is trying to do an external query?
Isn't this only related to some missing data in the test01.data file?

**Comment by vlevigneron** [comment]
@mtoma yes I guess this is the point. With new Engine version, caching of queries is different, that's why I had to create again all .data files in t/ repository of Engine. If you need that kind of file for Backend, you need to create a new one and push it in the repo.

**Comment by sandoche2k** [comment]
@mtoma to investigate the blib stuff for verifying whether two ZM engine versions.

**Comment by mattias-p** [comment]
For the record I'm able to reproduce this on my development machine (Debian 8).
When running with ZONEMASTER_RECORD=1 the test consistently passes. When running with ZONEMASTER_RECORD=0 the test consistently fails.

Also note that test01.data might not be compatible with whatever version of Engine you're running. In my current setup (current Engine develop branch) the test still fails if I re-record the data file. But interestingly, if I do it fails faster.

**Comment by matsduf** [comment]
This should be solved by #492
